packages/frontend-core/src/components/grid/stores/datasources/table.ts

```diff
@@ -1,32 +1,55 @@
-import { SortOrder } from "@budibase/types"
+import {
+  Row,
+  SaveRowRequest,
+  SaveRowResponse,
+  SaveTableRequest,
+  SortOrder,
+} from "@budibase/types"
 import { get } from "svelte/store"
+import { Store as StoreContext } from ".."
 
 const SuppressErrors = true
 
-export const createActions = context => {
+interface TableActions {
+  table: {
+    actions: {
+      saveDefinition: (newDefinition: SaveTableRequest) => Promise<void>
+      addRow: (row: SaveRowRequest) => Promise<SaveRowResponse>
+      updateRow: (row: SaveRowRequest) => Promise<SaveRowResponse>
+      deleteRows: (rows: (string | Row)[]) => Promise<void>
+      getRow: (id: string) => Promise<Row>
+      isDatasourceValid: (datasource: { type: string; tableId: any }) => boolean
```

Comment:
Similar to previous point, would be good to have a type for this that they all use (even for us to replace/update in future)